api/src/auth/auth.service.ts

```diff
@@ -0,0 +1,72 @@
+import { ExistingUserDTO } from './../user/dtos/existing-user.dto';
+import { UserDetails } from './../user/user-details.interface';
+import { Injectable } from '@nestjs/common';
+
+import * as bcrypt from 'bcrypt';
+import { NewUserDTO } from 'src/user/dtos/new-user.dto';
+
+import { UserService } from './../user/user.service';
+import { JwtService } from '@nestjs/jwt';
+
+@Injectable()
+export class AuthService {
+  constructor(
+    private userService: UserService,
+    private jwtService: JwtService,
+  ) {}
+
+  async hashPassword(password: string): Promise<string> {
+    return bcrypt.hash(password, 12);
+  }
+
+  async register(user: Readonly<NewUserDTO>): Promise<UserDetails | any> {
+    const { name, email, password } = user;
+
+    const existingUser = await this.userService.findByEmail(email);
+
+    if (existingUser) return 'Email taken!';
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider structured error response for email taken</b></div>
<div id="fix">

Consider returning a more specific error message or throwing a custom exception instead of a string. This would provide better error handling and allow for more precise error responses.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -27,1 +27,1 @@
 -    if (existingUser) return 'Email taken!';
 +    if (existingUser) throw new ConflictException('Email already in use');
```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/omkardongre/amazon-clone/pull/2#issuecomment-2466146244>#18f66c</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged